File: DSA/Anisha_Parveen/Day_10_Ques_1.cpp
#include <iostream>
#include <queue>

using namespace std;

void reverseQueue(queue<int>& q) {
    if (q.empty() || q.size() == 1) {
        return;
    }
    
    int front = q.front();
    q.pop();
    
    reverseQueue(q);
    
    q.push(front);
}

void printQueue(queue<int> q) {
    while (!q.empty()) {
        cout << q.front() << " ";
        q.pop();
    }
    cout << endl;
}

int main() {
    queue<int> q;
    q.push(1);
    q.push(2);
    q.push(3);
    
    cout << "Original Queue: ";
    printQueue(q);
    
    reverseQueue(q);
    
    cout << "Reversed Queue: ";
    printQueue(q);
    
    return 0;
}

File: DSA/Asmi_Garg/day10_ques1.java
import java.util.LinkedList;
import java.util.Queue;

public class ReverseQueue {
    
    public static void reverseQueue(Queue<Integer> queue) {
        if (queue.isEmpty()) {
            return;
        }

        int front = queue.poll();
        reverseQueue(queue);
        queue.offer(front);
    }

    public static void printQueue(Queue<Integer> queue) {
        for (Integer element : queue) {
            System.out.print(element + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Queue<Integer> queue = new LinkedList<>();
        queue.offer(1);
        queue.offer(2);
        queue.offer(3);

        System.out.println("Original Queue:");
        printQueue(queue);

        reverseQueue(queue);

        System.out.println("Reversed Queue:");
        printQueue(queue);
    }
}
